tsconfig.typecheck.json

```diff
@@ -6,7 +6,10 @@
       "path": "docs/tsconfig.json"
     },
     {
-      "path": "packages/mui-base/tsconfig.build.json"
+      "path": "packages/mui-base/tsconfig.json"
+    },
+    {
+      "path": "packages/mui-base/tsconfig.test.json"
```

Comment:
Been thinking about this a bit. If vscode looks at the root tsconfig.json to set up its type checking, wouldn't it make more sense to make sure that it references all projects. And instead use a separate tsconfig for type generation?

Reply:
As discussed on the meeting, I changed it to have a separate config for tests and production code and an umbrella one that references both. This main one is called tsconfig.json and can be used by VSCode.

A slight drawback of this setup is that now the tests need to output their definition files somewhere. I created a build-test directory for it.